hoomd/md/pytest/test_potential.py

```diff
@@ -1134,3 +1134,75 @@ def test_pickling(simulation_factory, two_particle_snapshot_factory,
     sim.operations.integrator = integrator
     sim.run(0)
     pickling_check(pot)
+
+
+@pytest.mark.parametrize("mode", ['none', 'shift', 'xplor'])
+def test_shift_mode_with_lrc(simulation_factory, two_particle_snapshot_factory,
+                             mode):
+    cell = md.nlist.Cell(buffer=0.4)
+    lj = md.pair.LJ(nlist=cell,
+                    default_r_cut=2.5,
+                    mode=mode,
+                    tail_correction=True)
+
+    lj.params[('A', 'A')] = {'sigma': 1, 'epsilon': 0.5}
+    snap = two_particle_snapshot_factory(dimensions=3, d=.5)
+    sim = simulation_factory(snap)
+    integrator = md.Integrator(dt=0.005)
+    integrator.forces.append(lj)
+    integrator.methods.append(
+        hoomd.md.methods.Langevin(hoomd.filter.All(), kT=1))
+    sim.operations.integrator = integrator
+    shift_allowed_modes = {'none'}
+    if mode not in shift_allowed_modes:
+        with pytest.raises(RuntimeError):
+            sim.run(1)
+    else:
+        sim.run(1)
+
+
+def test_lrc_non_lj(simulation_factory, two_particle_snapshot_factory):
+    # test we can't pass in tail_correction to non-LJ pair potential
+    cell = md.nlist.Cell(buffer=0.4)
+    with pytest.raises(TypeError):
+        # flake8 complains about unused variable with gauss = md.pair.Gauss(...)
+        md.pair.Gauss(nlist=cell,
+                      default_r_cut=2.5,
+                      mode='none',
+                      tail_correction=True)
+
+
+def test_tail_corrections(simulation_factory, two_particle_snapshot_factory):
```

Comment:
Use an `assert_allclose` regression check on the values of the energy and pressure.